File: LeetCode/Simulation/Python/Leetcode1441-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        ans,j = [], 0
        for i in range(1,n + 1):
            if target[j] == i:
                j += 1
                ans.append("Push")
            else:
                ans.append("Push")
                ans.append("Pop")
            if i == target[-1]:
                break
        return ans

File: LeetCode/Simulation/JS/Leetcode1441-build-an-array-with-stack-operations.js
var buildArray = function (target, n) {
  let [ans, j] = [[], 0];
  for (let i = 1; i <= n; i++) {
    if (i == target[j]) {
      j++;
      ans.push("Push");
    } else {
      ans.push("Push");
      ans.push("Pop");
    }
    if (i == target[target.length - 1]) {
      break;
    }
  }
  return ans;
};